src/kps/business/RouteCalculator.java

```diff
@@ -10,12 +10,13 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class RouteCalculator {
     private List<Route> routes;
 
     public RouteCalculator(Simulation simulation) {
-        this.routes = simulation.getRoutes();
+        this.routes = simulation.getUniqueRoutes().stream().filter(r -> !r.isDiscontinued()).collect(Collectors.toList());
```

Comment:
I'm looking at the getUniqueRoutes method, shouldn't if(... !r1.getCompany().equals(r2.getCompany()) ...) { continue; } by if equals to instead of if not equals to?